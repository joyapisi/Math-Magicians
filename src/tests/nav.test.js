import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Nav from '../components/nav';

describe('Nav', () => {
  test('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should navigate to the correct routes', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>,
    );

    const homeLink = getByText('Home');
    const calculatorLink = getByText('Calculator');
    const quotesLink = getByText('Quotes');

    expect(homeLink.getAttribute('href')).toBe('/');
    expect(calculatorLink.getAttribute('href')).toBe('/Calculator');
    expect(quotesLink.getAttribute('href')).toBe('/Quotes');
  });
});
