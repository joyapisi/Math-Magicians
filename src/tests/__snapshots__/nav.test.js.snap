// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render correctly 1`] = `
<nav>
  <h2>
    Math Magicians
  </h2>
  <div
    id="nav-box"
  >
    <a
      href="/"
    >
      Home
    </a>
    <span
      id="seperator"
    >
      |
    </span>
    <a
      href="/Calculator"
    >
      Calculator
    </a>
    <span
      id="seperator"
    >
      |
    </span>
    <a
      href="/Quotes"
    >
      Quotes
    </a>
  </div>
</nav>
`;
